pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -253,18 +261,37 @@ func (le *LeaderEngine) Subscribe() <-chan struct{} {
 	return c
 }
 
-// GetLeaderElectionRecord is used in for the Flare and for the Status commands.
-func GetLeaderElectionRecord() (leaderDetails rl.LeaderElectionRecord, err error) {
-	var led rl.LeaderElectionRecord
-	client, err := apiserver.GetAPIClient()
+func CanUseLease(client discovery.DiscoveryInterface) (bool, error) {
+	resourceList, err := client.ServerResourcesForGroupVersion("coordination.k8s.io/v1")
```

Comment:
Can you check if it can return this error: https://github.com/DataDog/datadog-agent/blob/c7d23930514c809796cd5b20dcf16776d4b1eedc/pkg/collector/corechecks/cluster/ksm/kubernetes_state.go#L319-L322

It happens when the discovery client in the Cluster Agent lists resources but one of the resource is `externalMetrics`, which is failing because the Cluster Agent is starting.... waiting on the discovery client.

Reply:
I planned to replace this function by `IsResourceEnabled` as soon as we bump kubernetes version to 1.26.2.
Also, I don't think it can return this error. Looking at `ServerGroupsAndResources` code, the error is explicit: https://github.com/kubernetes/client-go/blob/6c514ab3ffc5f500fd373f09956780c0e3bbe028/discovery/discovery_client.go#L456

The function I use does not return such error:
https://github.com/kubernetes/client-go/blob/6c514ab3ffc5f500fd373f09956780c0e3bbe028/discovery/discovery_client.go#L347